testing/src/cypress/page-objects/Institution-objects/DashboardInstitutionObject.ts

```diff
@@ -44,6 +137,122 @@ export default class DashboardInstitutionObject extends BaseMethods {
   }
 
+  institutionProgramsHeader() {
+    return this.getElementByCyId("programsHeader");
+  }
+
+  viewProgramButton() {
+    return this.getElementByCyId("viewProgram");
+  }
+
+  institutionSearchInputBox() {
+    return this.getElementByCyId("searchBox").clear();
+  }
+
+  createInstitutionProgram() {
+    return this.getElementByCyId("createProgram");
+  }
+
+  institutionProgramName() {
+    return this.getElementByCyId("programName");
+  }
+
+  institutionProgramCIPInputText() {
+    return this.getElementByCyId("cipCode");
+  }
+
+  institutionProgramCredential() {
+    return this.getElementByCyId("programCredential");
+  }
+
+  institutionProgramStatus() {
+    return this.getElementByCyId("programStatus");
+  }
+
+  institutionProgramStudyPeriods() {
+    return this.getElementByCyId("programStudyPeriods");
+  }
+
+  selectCredentialType(credentialType: CredentialTypes) {
+    this.getElementByCyId("credentialType").select(credentialType);
+  }
+
+  selectProgramLength(programLength: ProgramLengthOptions) {
+    this.getElementByCyId("completionYears").select(programLength);
+  }
+
+  selectRegulatoryBody(regulatoryBody: RegulatoryBodyOptions) {
+    this.getElementByCyId("regulatoryBody").select(regulatoryBody);
+  }
+
+  studentPartTimeBasisRadioButton(checkBoxToSelect: ProgramIntensityOptions) {
+    this.selectCheckBox("programIntensity", checkBoxToSelect);
+  }
+
+  deliverabilityOnsiteRadioButton(checkBoxToSelect: ProgramDeliveryOptions) {
+    this.selectCheckBox("programDeliveryTypes", checkBoxToSelect);
+  }
+
+  deliveredOnlineAlsoRadioButton(
+    checkBoxToSelect: DeliveredOnlineAlsoOnsiteOptions
+  ) {
+    this.selectCheckBox("deliveredOnlineAlsoOnsite", checkBoxToSelect);
+  }
+
+  courseLoadCalculationRadioButton(
+    checkBoxToSelect: CourseLoadCalculationOptions
+  ) {
+    this.selectCheckBox("courseLoadCalculation", checkBoxToSelect);
+  }
+
+  entranceRequirementsRadioButton(
+    checkBoxToSelect: EntranceRequirementsOptions
+  ) {
+    this.selectCheckBox("entranceRequirements", checkBoxToSelect);
+  }
+
+  eslEligibilityRadioButton(checkBoxToSelect: EslEligibilityOptions) {
+    this.selectCheckBox("eslEligibility", checkBoxToSelect);
+  }
+
+  hasJointInstitutionRadioButton(checkBoxToSelect: JointInstitutionOptions) {
+    this.selectCheckBox("hasJointInstitution", checkBoxToSelect);
+  }
+
+  hasJointDesignatedInstitutionRadioButton(
+    checkBoxToSelect: JointDesignatedInstitution
+  ) {
+    this.selectCheckBox("hasJointDesignatedInstitution", checkBoxToSelect);
+  }
+
+  hasWILComponentRadioButton(checkBoxToSelect: WILComponentOptions) {
+    this.selectCheckBox("hasWILComponent", checkBoxToSelect);
+  }
+
+  isWILApprovedRadioButton(checkBoxToSelect: WILApprovalOptions) {
+    this.selectCheckBox("isWILApproved", checkBoxToSelect);
+  }
+
+  hasTravelRadioButton(checkBoxToSelect: TravelProgramEligibilityOptions) {
+    this.selectCheckBox("hasTravel", checkBoxToSelect);
+  }
+
+  fieldOfStudyCode() {
```

Comment:
cy id?

Reply:
if you are referring to `this.selectCheckBox` - thats new function which checks for the checkBox types which are identified by the data-cy and selects the option provided in the arg.

Reply:
Sorry I was talking about line 241. 
I misread it as `getElementById` instead of `getElementBycyId`. 
Please Ignore it.